conjure-undertow-processor/src/test/java/com/palantir/conjure/java/undertow/processor/sample/ContextParamNameClash.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.processor.sample;
+
+import com.palantir.conjure.java.undertow.annotations.Handle;
+import com.palantir.conjure.java.undertow.annotations.HttpMethod;
+import com.palantir.conjure.java.undertow.lib.RequestContext;
+
+public interface ContextParamNameClash {
+
+    @SuppressWarnings("checkstyle:ParameterName")
+    @Handle(method = HttpMethod.GET, path = "/context")
+    String context(@Handle.QueryParam("requestContext") String requestContext_, RequestContext context);
```

Comment:
Worth testing something like this as well:

```java
    @Handle(method = HttpMethod.GET, path = "/{exchange}")
    String exchange(@Safe @Handle.PathParam String exchange);
```

Reply:
Yep, this will also fail. Should we also rename our internal `HttpServerExchange` to `exchange_` to avoid clashes in most cases?

Reply:
Updated the internal name to `exchange_` and added a failing test.

Reply:
I could go either way. On one hand, using the `exchange` name is nice for readability, and we can rename parameters around it for consistency if they happen to crash. On the other hand, that's more work.

Reply:
Let's keep exchange as the internal param name for now. It's trivial and safe to change once someone runs into this problem but maybe we'll just never encounter it. Updated.